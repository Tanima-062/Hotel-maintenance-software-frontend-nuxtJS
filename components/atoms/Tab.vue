<template>
  <div class="relative">
    <button
      class="border border-solid border-bdColor px-4 py-3"
      :class="styles"
      @click="click"
    >
      <slot />
    </button>
    <div
      v-if="isActive"
      class="border border-solid border-btnColor bg-btnColor absolute top-0 left-0 w-full h-4px"
    />
    <div
      v-if="isActive"
      class="bg-white absolute bottom-0 left-0 wfull h-px -mb-px ml-px"
    />
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component({})
export default class extends Vue {
  @Prop({ default: false })
  isActive!: boolean

  @Prop({ default: '' })
  tabWidth!: string

  @Prop({ default: false })
  disable!: boolean

  get styles(): string {
    const classList = []
    classList.push(
      this.isActive
        ? 'border-b-0 text-btnColor bg-white'
        : 'text-black bg-tabBgColor'
    )
    classList.push(this.tabWidth)
    if (this.disable) {
      classList.push('text-bdColor cursor-not-allowed')
    }
    return classList.join(' ')
  }

  @Emit()
  private click(e: Event): Event {
    return e
  }
}
</script>
<style lang="sass" scoped>
.wfull
  width: calc(100% - 2px)
</style>
