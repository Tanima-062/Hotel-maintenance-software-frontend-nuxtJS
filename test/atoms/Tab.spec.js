import { shallowMount } from '@vue/test-utils'
import Tab from '@/components/atoms/Tab.vue'

describe('Atoms Tab Check', () => {
  test('Tab Test', () => {
    const wrapper = shallowMount(Tab)

    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
