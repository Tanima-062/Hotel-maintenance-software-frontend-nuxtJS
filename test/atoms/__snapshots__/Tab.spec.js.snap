// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms Tab Check Tab Test 1`] = `
<div
  class="relative"
>
  <button
    class="border border-solid border-bdColor px-4 py-3 text-black bg-tabBgColor "
  />
   
  <!---->
   
  <!---->
</div>
`;
